src/Shared/RoslynUtils/ParsabilityHelper.cs

```diff
@@ -16,50 +16,65 @@ namespace Microsoft.AspNetCore.Analyzers.Infrastructure;
 
 internal static class ParsabilityHelper
 {
-    private static bool IsTypeAlwaysParsableOrBindable(ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes)
+    private static bool IsTypeAlwaysParsableOrBindable(ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes, out ParsabilityMethod? parsabilityMethod)
     {
         // Any enum is valid.
         if (typeSymbol.TypeKind == TypeKind.Enum)
         {
+            parsabilityMethod = ParsabilityMethod.Enum;
             return true;
         }
 
         // Uri is valid.
         if (SymbolEqualityComparer.Default.Equals(typeSymbol, wellKnownTypes.Get(WellKnownType.System_Uri)))
         {
+            parsabilityMethod = ParsabilityMethod.Uri;
             return true;
         }
 
         // Strings are valid.
         if (typeSymbol.SpecialType == SpecialType.System_String)
         {
+            parsabilityMethod = ParsabilityMethod.String;
             return true;
         }
 
+        parsabilityMethod = null;
         return false;
     }
 
     internal static Parsability GetParsability(ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes)
     {
-        if (IsTypeAlwaysParsableOrBindable(typeSymbol, wellKnownTypes))
+        return GetParsability(typeSymbol, wellKnownTypes, out var _);
+    }
+
+    internal static Parsability GetParsability(ITypeSymbol typeSymbol, WellKnownTypes wellKnownTypes, out ParsabilityMethod? parsabilityMethod)
+    {
+        if (IsTypeAlwaysParsableOrBindable(typeSymbol, wellKnownTypes, out parsabilityMethod))
         {
             return Parsability.Parsable;
         }
 
         // MyType : IParsable<MyType>()
         if (IsParsableViaIParsable(typeSymbol, wellKnownTypes))
         {
+            parsabilityMethod = ParsabilityMethod.IParsable;
             return Parsability.Parsable;
         }
 
         // Check if the parameter type has a public static TryParse method.
         var tryParseMethods = typeSymbol.GetMembers("TryParse").OfType<IMethodSymbol>();
-        foreach (var tryParseMethodSymbol in tryParseMethods)
+
+        if (tryParseMethods.Any(m => IsTryParseWithFormat(m, wellKnownTypes)))
         {
-            if (IsTryParse(tryParseMethodSymbol) || IsTryParseWithFormat(tryParseMethodSymbol, wellKnownTypes))
-            {
-                return Parsability.Parsable;
-            }
+            parsabilityMethod = ParsabilityMethod.TryParseWithFormatProvider;
+            return Parsability.Parsable;
+        }
+
+        if (tryParseMethods.Any(m => IsTryParse(m)))
```

Comment:
```suggestion
        if (tryParseMethods.Any(IsTryParsem))
```

To appease the build 🙇🏽